### PR TITLE
#1465 Allow camel case service names if no Helm-chart is provided

### LINKED ISSUE #1465: Allow case-sensitive service names in "keptn create service"
In `keptn onboard service` and `keptn create service` we only allow service names, which only contain lower case letters. This is due to the fact that the k8s service name has to match Keptn service names. 
However, if we use quality gates only, we do not manage k8s service names and, hence, we should allow to use case-sensitive service names in `keptn create service`.

### PR DESCRIPTION
**Pull request by agrimmer**
Closes #1465

### CODE REVIEW

**Comment by agrimmer** [comment]
@christian-kreuzberger-dtx May I also ask you for your feedback?

File: cli/cmd/create_service.go [hunk 1]
```diff
@@ -40,9 +40,8 @@ Example:
 			return errors.New("required argument SERVICENAME not set")
 		}
 
-		if !keptnutils.ValidateKeptnEntityName(args[0]) {
-			errorMsg := "Service name contains upper case letter(s) or special character(s).\n"
-			return errors.New(errorMsg)
+		if !keptnutils.ValididateUnixDirectoryName(args[0]) {
+			return errors.New("Service name contains special character(s)")
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/create_service.go]: Maybe specify that the service name contains unsupported characters and add some information (e.g., via a hyperlink?) about those.

File: helm-service/controller/onboarder.go [hunk 2]
```diff
@@ -124,18 +124,18 @@ func (o *Onboarder) DoOnboard(ce cloudevents.Event, loggingDone chan bool) error
...
-	errorMsg := "Service name contains upper case letter(s) or special character(s).\n " +
-		"Keptn relies on the following conventions: " +
-		"start with a lower case letter, then lower case letters, numbers, and hyphens are allowed."
-
 	if event.HelmChart == "" {
 		// Case when only a service is created but not onboarded (i.e. no Helm chart is available)
-		if !keptnutils.ValidateKeptnEntityName(event.Service) {
-			return errors.New(errorMsg)
+		if !keptnutils.ValididateUnixDirectoryName(event.Service) {
+			return errors.New("Service name contains special character(s)")
```
Comment (christian-kreuzberger-dtx) [inline helm-service/controller/onboarder.go]: Same as with the CLI, please add some more information and call it "unsupported characters"

File: cli/cmd/create_service.go (hunk 1 above)
Comment (agrimmer) [reply cli/cmd/create_service.go]: I improved the error message.

File: helm-service/controller/onboarder.go (hunk 2 above)
Comment (agrimmer) [reply helm-service/controller/onboarder.go]: I improved the error message.

**Pull request merged**